File: solutions/ocaml/019-merge-two-sorted-lists.ml
(*
  Problem 19: Merge Two Sorted Lists (LeetCode 21)
  Difficulty: Easy
  Language: OCaml
*)
let rec merge_two_lists l1 l2 =
  match l1, l2 with
  | [], ys -> ys
  | xs, [] -> xs
  | x :: xs, y :: ys ->
    if x <= y then x :: merge_two_lists xs (y :: ys)
    else y :: merge_two_lists (x :: xs) ys

File: solutions/elixir/019-merge-two-sorted-lists.ex
# Problem 19: Merge Two Sorted Lists (LeetCode 21)
# Difficulty: Easy
# Language: Elixir
# 
defmodule MergeLists do
  def merge_two_lists([], l2), do: l2
  def merge_two_lists(l1, []), do: l1
  def merge_two_lists([h1 | t1] = l1, [h2 | t2] = l2) do
    if h1 <= h2 do
      [h1 | merge_two_lists(t1, l2)]
    else
      [h2 | merge_two_lists(l1, t2)]
    end
  end
end
